src/app/users/view-user/view-user.component.ts

```diff
@@ -58,4 +59,23 @@ export class ViewUserComponent implements OnInit {
     });
   }
 
+  /**
+   * Change Password of the Users.
+   */
+  changeUserPassword() {
+    const changeUserPasswordDialogRef = this.dialog.open(ChangePasswordDialogComponent, {
+      width: '400px',
+      height: '300px'
+    });
+    changeUserPasswordDialogRef.afterClosed().subscribe((response: any) => {
+      if (response.password && response.repeatPassword) {
+        const password = response.password;
+        const repeatPassword = response.repeatPassword;
+        this.usersService.changePassword(this.userData.id, {password: password, repeatPassword: repeatPassword}).subscribe(() => {
```

Comment:
Please follow the documentation to send the data to the API [Link](https://demo.mifos.io/api-docs/apiLive.htm#users_update).

You need to add another parameter, `firstname` in the data to make a successful request. And also make another object named `data` with all the fields and you can directly send that to the API.

Reply:
Done :) Please review